pkg/placement/membership.go

```diff
@@ -307,11 +313,55 @@ func (p *Service) performTableDissemination() {
 func (p *Service) performTablesUpdate(hosts []placementGRPCStream, newTable *v1pb.PlacementTables) {
 	// TODO: error from disseminationOperation needs to be handle properly.
 	// Otherwise, each Dapr runtime will have inconsistent hashing table.
+	startedAt := time.Now()
+
+	var wg sync.WaitGroup
 	for _, host := range hosts {
-		p.disseminateOperation([]placementGRPCStream{host}, "lock", nil)
-		p.disseminateOperation([]placementGRPCStream{host}, "update", newTable)
-		p.disseminateOperation([]placementGRPCStream{host}, "unlock", nil)
+		wg.Add(1)
```

Comment:
I wonder if this can cause issues for high scale clusters?  I think we should add a `parallelism` setting to make sure we don't spawn too many go routines at once. We can add it as a separate PR since this one makes things better already.

Reply:
I'm not overly concerned here, hundreds of goroutines (representing hundreds of nodes, a very unlikely and uncommon deployment topology) would be fine IMO. a goroutine takes 2kb of memory and so memory usage would go up minimally, while CPU usage should increase but not to unsustainable levels as the IO operation in dissemination uses already established streams and not creating/destroying new connections.